Avoid creating executable files

ramikg opened the pull request:
### Description
This PR removes all executable bits in calls to `os.OpenFile()` or `os.WriteFile()`.

Previously, the file mode passed to these functions was `os.ModePerm` (which is equal to 0777, i.e. read/write/execute for everyone), which may have undesirable consequences.  
This value is sometimes used by Go programmers because it is conveniently exported in the `os` package, but it's intended usage is as a bitmask, and not as a file mode.  
I've changed this value to 0666 (i.e. removed the executable bits), like the one used by Go's `os.Create()`.

### Checklist
- [x] Code compiles correctly
- [x] Run ``make fmt`` to fix inconsistent formats
- [x] Run ``make lint`` to get lint errors and fix all of them
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing
- [x] Extended the README / documentation, if necessary

ramikg commented:
I have read the CLA Document and I hereby sign the CLA

sfc-gh-pfus commented on azure_storage_client.go:
```diff
@@ -204,7 +204,7 @@ func (util *snowflakeAzureClient) uploadFile(
 	} else {
 		var f *os.File
-		f, err = os.OpenFile(dataFile, os.O_RDONLY, os.ModePerm)
+		f, err = os.OpenFile(dataFile, os.O_RDONLY, 0666)
```
Can we introduce a const instead of repeating the value everywhere? I would place it in `file_util.go` and name `ModeReadWrite`.

sfc-gh-pfus replied:
What would you think about boy scout rule and replacing `os.OpenFile` with `os.Open`? We don't need permission here as we don't create a file.

sfc-gh-pfus commented on doc.go:
```diff
@@ -889,7 +889,7 @@ both an escape character and as a separator in path names.
 (The ReplaceAll() function is needed on Windows to handle backslashes in the path to the file.)
 
-	fileStream, _ := os.OpenFile(fname, os.O_RDONLY, os.ModePerm)
+	fileStream, _ := os.OpenFile(fname, os.O_RDONLY, 0666)
```
Maybe here we can also introduce `os.Open` as we don't modify the file?

sfc-gh-pfus commented on encrypt_util.go:
```diff
@@ -232,7 +232,7 @@ func decryptFile(
 	}
 	defer tmpOutputFile.Close()
-	infile, err := os.OpenFile(filename, os.O_RDONLY, os.ModePerm)
+	infile, err := os.OpenFile(filename, os.O_RDONLY, 0666)
```
`os.Open`?

sfc-gh-pfus commented on encrypt_util_test.go:
```diff
@@ -33,7 +33,7 @@ func TestEncryptDecryptFile(t *testing.T) {
 	inputFile := "test_encrypt_decrypt_file"
 
-	fd, err := os.OpenFile(inputFile, os.O_CREATE|os.O_WRONLY, os.ModePerm)
+	fd, err := os.OpenFile(inputFile, os.O_CREATE|os.O_WRONLY, 0666)
```
`os.Create`?

ramikg replied:
A general comment about replacing all `os.OpenFile(..., os.O_CREATE, 0666)` with `os.Create`:  
It will affect behavior in a potentially dangerous way, as after the change it will truncate the file when it doesn't exist (i.e. delete its contents).
There's a good chance it won't have any negative effect, but I'd still check it on a case-by-case basis.

sfc-gh-pfus replied:
Yes, you are right. But I tried to focus only on tests :) I didn't mention any `os.Create` related lines in production code.

ramikg replied:
I've changed to `os.Create` in test files.

sfc-gh-pfus commented on encrypt_util_test.go:
```diff
@@ -54,7 +54,7 @@ func TestEncryptDecryptFile(t *testing.T) {
 	defer os.Remove(decryptedFile)
 
-	fd, err = os.OpenFile(decryptedFile, os.O_RDONLY, os.ModePerm)
+	fd, err = os.OpenFile(decryptedFile, os.O_RDONLY, 0666)
```
`os.Open`?

sfc-gh-pfus commented on encrypt_util_test.go:
```diff
@@ -168,7 +168,7 @@ func generateKLinesOfNByteRows(numLines int, numBytes int, tmpDir string) (strin
 	}
 	fname := path.Join(tmpDir, "file"+strconv.FormatInt(int64(numLines*numBytes), 10))
-	f, err := os.OpenFile(fname, os.O_CREATE|os.O_WRONLY, os.ModePerm)
+	f, err := os.OpenFile(fname, os.O_CREATE|os.O_WRONLY, 0666)
```
`os.Create`?

sfc-gh-pfus commented on file_util.go:
```diff
@@ -39,12 +39,12 @@ func (util *snowflakeFileUtil) compressFileWithGzip(fileName string, tmpDir stri
 	gzipFileName := filepath.Join(tmpDir, basename+"_c.gz")
 
-	fr, err := os.OpenFile(fileName, os.O_RDONLY, os.ModePerm)
+	fr, err := os.OpenFile(fileName, os.O_RDONLY, 0666)
```
`os.Open`?

sfc-gh-pfus commented on local_storage_client_test.go:
```diff
@@ -73,7 +73,7 @@ func TestLocalUpload(t *testing.T) {
 		t.Fatal("overwrite is false. should have skipped")
 	}
-	fileStream, _ := os.OpenFile(fname, os.O_RDONLY, os.ModePerm)
+	fileStream, _ := os.OpenFile(fname, os.O_RDONLY, 0666)
```
`os.Open`?

sfc-gh-pfus commented on put_get_test.go:
```diff
@@ -392,14 +392,14 @@ func testPutGet(t *testing.T, isStream bool) {
 		dbt.mustExec("create or replace table " + tableName +
 			" (a int, b string)")
-		fileStream, err := os.OpenFile(fname, os.O_RDONLY, os.ModePerm)
+		fileStream, err := os.OpenFile(fname, os.O_RDONLY, 0666)
```
`os.Open`?

sfc-gh-pfus commented on put_get_test.go:
```diff
@@ -517,15 +517,15 @@ func TestPutGetGcsDownscopedCredential(t *testing.T) {
 		dbt.mustExec("create or replace table " + tableName +
 			" (a int, b string)")
-		fileStream, err := os.OpenFile(fname, os.O_RDONLY, os.ModePerm)
+		fileStream, err := os.OpenFile(fname, os.O_RDONLY, 0666)
```
`os.Open`?

sfc-gh-pfus commented on put_get_user_stage_test.go:
```diff
@@ -69,7 +69,7 @@ func putGetUserStage(t *testing.T, tmpDir string, numberOfFiles int, numberOfLin
 		var fs *os.File
 		if isStream {
-			fs, _ = os.OpenFile(files, os.O_RDONLY, os.ModePerm)
+			fs, _ = os.OpenFile(files, os.O_RDONLY, 0666)
```
`os.Open`?

sfc-gh-pfus reviewed (commented):
Hello @ramikg ! We are really thankful for your initiative! If you would like to implement boy scout rule and address some of my comments it would be really great. Thanks!

ramikg commented:
> Hello @ramikg ! We are really thankful for your initiative! If you would like to implement boy scout rule and address some of my comments it would be really great. Thanks!

Thank you. I've addressed all comments except for one, which I've commented on.

sfc-gh-pfus commented:
Thanks a lot! I answered your concern, I hope that I can convince you.

sfc-gh-pfus commented:
@ramikg thank you very much for you input! Your help is appreciated :) Unfortunately, our policy does not allow direct merges from community, but I marked you as `co-authored-by` in the git commit.
